GooglePlayInstant/Editor/ApkSigner.cs

```diff
@@ -0,0 +1,184 @@
+                        : Environment.GetEnvironmentVariable("HOME");
+                if (string.IsNullOrEmpty(homePath))
+                {
+                    Debug.LogErrorFormat("Failed to locate home directory that contains {0}.", AndroidDebugKeystore);
+                    return false;
+                }
+
+                keystoreName = Path.Combine(homePath, AndroidDebugKeystore);
+                keystorePass = "android";
+                keyaliasName = "androiddebugkey";
+                keyaliasPass = "android";
+            }
+            else
+            {
+                keystoreName = PlayerSettings.Android.keystoreName;
+                keystorePass = PlayerSettings.Android.keystorePass;
+                keyaliasName = PlayerSettings.Android.keyaliasName;
+                keyaliasPass = PlayerSettings.Android.keyaliasPass;
+            }
+
+            var arguments = string.Format(
+                "-jar {0} sign --ks {1} --ks-key-alias {2} --pass-encoding utf-8 {3}",
+                GetApkSignerJarPath(), keystoreName, keyaliasName, apkPath);
+
+            var promptToPasswordDictionary = new Dictionary<string, string>
+            {
+                // Example keystore password prompt: "Keystore password for signer #1: "
+                {"Keystore password for signer", keystorePass},
+                // Example keyalias password prompt: "Key \"androiddebugkey\" password for signer #1: "
+                {"password for signer", keyaliasPass}
+            };
+            var responder = new ApkSignerResponder(promptToPasswordDictionary);
+            var result = CommandLine.Run(JavaUtilities.JavaBinaryPath, arguments, ioHandler: responder.AggregateLine);
+            if (result.exitCode == 0)
+            {
+                return true;
+            }
+
+            Debug.LogErrorFormat("\"java {0}\" failed with exit code {1}", arguments, result.exitCode);
+            return false;
+        }
+
+        private static string GetApkSignerJarPath()
+        {
+            var newestBuildToolsVersion = AndroidBuildTools.GetNewestBuildToolsVersion();
+            if (newestBuildToolsVersion == null)
+            {
+                return null;
+            }
+
+            var newestBuildToolsPath = Path.Combine(AndroidBuildTools.GetBuildToolsPath(), newestBuildToolsVersion);
+            var apkSignerJarPath = Path.Combine(newestBuildToolsPath, "lib/apksigner.jar");
+            if (File.Exists(apkSignerJarPath))
+            {
+                return apkSignerJarPath;
+            }
+
+            Debug.LogErrorFormat("Failed to locate apksigner.jar at path: {0}", apkSignerJarPath);
+            return null;
+        }
+
+        /// <summary>
+        /// Checks apksigner's stdout for password prompts and outputs the associated password to apksigner's stdin.
+        /// This is more secure than using apksigner's support for command line and file-based password input.
```

Comment:
I think we should elaborate on this comment, why is it more secure to pipe the password into stdin rather than passing it around as a CLI argument